src/main/scala/org/broadinstitute/dsde/workbench/sam/audit/SamAuditModel.scala

```diff
@@ -0,0 +1,120 @@
+package org.broadinstitute.dsde.workbench.sam.audit
+
+import org.broadinstitute.dsde.workbench.model.{WorkbenchException, WorkbenchGroupName, WorkbenchSubject, WorkbenchUserId}
+import org.broadinstitute.dsde.workbench.sam.model.{FullyQualifiedPolicyId, FullyQualifiedResourceId, ResourceAction, ResourceRoleName, ResourceTypeName}
+import spray.json.{DefaultJsonProtocol, DeserializationException, JsObject, JsString, JsValue, RootJsonFormat, SerializationException}
+
+import java.net.InetAddress
+
+sealed trait AuditEvent
+
+final case class AuditInfo(userId: Option[WorkbenchUserId], clientIp: Option[InetAddress])
+
+final case class ResourceChange(parent: Option[FullyQualifiedResourceId])
```

Comment:
This class is capturing what changed about a resource. In this case the only changeable thing I ended up with is the resource's parent. Note that the id here is for the parent, not the resource that is changing.

The idea behind all the Options in this file is that I don't want things showing up in json if they were not changed. So I agree that it is confusing to make parent an Option since it is the only thing that can change. But I don't think that will always be the case. Making it an Option is also consistent with AccessChange.